### PR TITLE
HPCC-15361 Use std::atomic for spin locks

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@mckellyln please review

File: system/jlib/jmutex.hpp [hunk 1]
```diff
@@ -373,21 +375,18 @@ class  SpinLock
...
     {   
-        owner.tid = 0;
-        nesting = 0;
-        atomic_set(&value, 0); 
     }
 #ifdef _DEBUG
     ~SpinLock()             
     { 
-        if (atomic_read(&value))
+        if (value)
```
Comment (mckellyln) [inline system/jlib/jmutex.hpp]: trivial - why value and not (just to be consistent) value.load(std::memory_order_xxx) ?

**Comment by mckellyln** [comment]
Excellent.

**Comment by richardkchapman** [comment]
@ghalliday Given your recent performance testing results, do you want me to hold off on merging this?

File: system/jlib/jmutex.hpp (hunk 1 above)
Comment (ghalliday) [inline system/jlib/jmutex.hpp]: I didn't because it is debug only code, and went for simplicity instead.

**Comment by ghalliday** [comment]
@richardkchapman I think my concerns are with atomic_cas (compare_exchange_xxx).  I don't think the other atomics have the same problems.

**Pull request merged**